src/components/AssetsStatusAlert/index.jsx

```diff
@@ -0,0 +1,228 @@
+          />
+        );
+        alertType = 'info';
+        break;
+      case assetActions.upload.UPLOAD_EXCEED_MAX_COUNT_ERROR:
+        alertDialog = (
+          <WrappedMessage
+            message={messages.assetsStatusAlertTooManyFiles}
+            values={{ max_count: (<FormattedNumber value={assetsStatus.maxFileCount} />) }}
+          />
+        );
+        alertType = 'danger';
+        break;
+      case assetActions.upload.UPLOAD_EXCEED_MAX_SIZE_ERROR:
+        alertDialog = (
+          <WrappedMessage
+            message={messages.assetsStatusAlertTooMuchData}
+            values={{ max_size: (<FormattedNumber value={assetsStatus.maxFileSizeMB} />) }}
+          />
+        );
+        alertType = 'danger';
+        break;
+      case assetActions.upload.UPLOAD_ASSET_FAILURE:
+        alertDialog = (
+          <WrappedMessage
+            message={messages.assetsStatusAlertGenericError}
+            values={{ assetName: assetsStatus.asset.name }}
+          />
+        );
+        alertType = 'danger';
+        break;
+      case assetActions.lock.TOGGLING_LOCK_ASSET_FAILURE:
+        alertDialog = (
+          <WrappedMessage
+            message={messages.assetsStatusAlertFailedLock}
+            values={{ assetName: assetsStatus.asset.name }}
+          />
+        );
+        alertType = 'danger';
+        break;
+      case assetActions.clear.CLEAR_FILTERS_FAILURE:
+        alertDialog = genericUpdateError;
+        alertType = 'danger';
+        break;
+      case assetActions.filter.FILTER_UPDATE_FAILURE:
+        alertDialog = genericUpdateError;
+        alertType = 'danger';
+        break;
+      case assetActions.paginate.PAGE_UPDATE_FAILURE:
+        alertDialog = genericUpdateError;
+        alertType = 'danger';
+        break;
+      case assetActions.sort.SORT_UPDATE_FAILURE:
+        alertDialog = genericUpdateError;
+        alertType = 'danger';
+        break;
+      default:
+        return;
+    }
+
+    this.setState({
+      statusAlertOpen: true,
+      statusAlertFields: {
+        alertDialog,
+        alertType,
+      },
+    });
+    this.statusAlertRef.focus();
+  }
+
+  closeDeleteStatus = () => {
+    this.closeStatusAlert();
+    this.props.onDeleteStatusAlertClose();
+  }
+
+  closeStatusAlert() {
+    this.props.clearAssetsStatus();
+
+    // clear out all status related state
+    this.setState({
+      statusAlertOpen: false,
+      statusAlertFields: {
+        alertDialog: '',
+        alertType: 'info',
+      },
+      uploadSuccessCount: 1,
+    });
+  }
+
+  updateUploadSuccessCount() {
+    const uploadSuccessCount = this.state.uploadSuccessCount + 1;
+    this.setState({
+      uploadSuccessCount,
+    });
+  }
+
+  render() {
+    const { assetsStatus } = this.props;
+
+    let onClose = this.closeStatusAlert;
+    if (Object.prototype.hasOwnProperty.call(assetActions.delete, assetsStatus.type)) {
+      onClose = this.closeDeleteStatus;
+    }
+
+    const statusAlert = (
```

Comment:
You could consider `return`ing directly here instead of assigning to a variable and then `return`ing that variable.